### PR TITLE
Support attaching multiple networks with vSphere VMs

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds support for configuring multiple networks for VMs on vSphere. The field vmNetName is considered deprecated in cluster and preset CRDs for vSphere and networks should be used instead. In case both network and vmNetName are specified for `clusters`, the validating webhooks will reject the cluster.

**Which issue(s) this PR fixes**:

xref https://github.com/kubermatic/dashboard/issues/6065

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* Support for configuring multiple networks for vSphere.
* [Action Required] The field `vmNetName` in `cluster` and `preset` CRDs is considered deprecated for vSphere and `networks` should be used instead.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -59,23 +59,22 @@ require (
 	github.com/spf13/pflag v1.0.6-0.20210604193023-d5e0c0615ace
 	github.com/stretchr/testify v1.8.4
 	github.com/vishvananda/netlink v1.2.1-beta.2.0.20220608195807-1a118fe229fc
-	github.com/vmware/go-vcloud-director/v2 v2.20.0
-	github.com/vmware/govmomi v0.30.5
+	github.com/vmware/go-vcloud-director/v2 v2.19.0
+	github.com/vmware/govmomi v0.30.2
```
Comment (mate4st) [inline go.mod]: Why do we downgrade some packages?

Comment (ahmedwaleedmalik) [reply go.mod]: I somehow messed them up when I rebased my PR. Should be fixed now.

**Comment by dharapvj** [comment]
Will there be similar change in Machine Controller as well? Currently, I see vmNetName present in MD configurations as well.

**Comment by ahmedwaleedmalik** [comment]
> Will there be similar change in Machine Controller as well? Currently, I see vmNetName present in MD configurations as well.

https://github.com/kubermatic/machine-controller/pull/1673

**Pull request merged**